modules/core_project_factory/main.tf

```diff
@@ -193,64 +206,50 @@ data "null_data_source" "default_service_account" {
   }
 }
 
-/******************************************
-  Default compute service account deletion
- *****************************************/
-resource "null_resource" "delete_default_compute_service_account" {
-  count = var.default_service_account == "delete" ? 1 : 0
+/************************************************************************************
+  Default compute service account action. Options are keep, depriviledge, delete.
+ ***********************************************************************************/
+resource "null_resource" "modify_default_compute_service_account" {
 
   provisioner "local-exec" {
-    command = "${path.module}/scripts/delete-service-account.sh ${google_project.main.project_id} ${data.null_data_source.default_service_account.outputs["email"]} ${var.credentials_path}"
-  }
-
-  triggers = {
-    default_service_account = data.null_data_source.default_service_account.outputs["email"]
-    activated_apis = join(",", var.activate_apis)
-  }
-
-  depends_on = [google_project_service.project_services]
-}
-
-/*********************************************
-  Default compute service account depriviledge
- ********************************************/
-resource "null_resource" "depriviledge_default_compute_service_account" {
-  count = var.default_service_account == "depriviledge" ? 1 : 0
-
-  provisioner "local-exec" {
-    command = "gcloud projects remove-iam-policy-binding ${google_project.main.project_id} --member='serviceAccount:${data.null_data_source.default_service_account.outputs["email"]}' --role='roles/editor'"
-
-    environment = {
-      CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE = var.credentials_path
-    }
+    command = <<EOD
+${path.module}/scripts/modify-service-account.sh \
+  --project_id='${google_project.main.project_id}' \
+  --sa_id='${data.null_data_source.default_service_account.outputs["email"]}' \
+  --credentials_path='${var.credentials_path}' \
+  --impersonate-service-account='${var.impersonate_service_account}' \
+  --action='${var.default_service_account}'
+EOD
   }
 
   triggers = {
     default_service_account = data.null_data_source.default_service_account.outputs["email"]
+    activated_apis          = join(",", var.activate_apis)
   }
 
   depends_on = [
     google_project_service.project_services,
     google_project_services.project_services_authority,
+    google_project_service.iam_credentials_api
   ]
 }
 
 /******************************************
   Default Service Account configuration
  *****************************************/
 resource "google_service_account" "default_service_account" {
-  account_id = "project-service-account"
+  account_id   = "project-service-account"
```

Comment:
Make sure to run the format/linting commands.

Reply:
No problem, this formatting change was the result of running `terraform fmt` FYI, running make -s has a bunch of errors locally for me for files I haven't touched, perhaps I am using a different version of shellcheck. Mine is 0.4.7, here is a sample of the errors:

```
In ./helpers/setup-sa.sh line 171:
\ \ - serviceAccount:${SA_ID}\\
^-- SC1117: Backslash is literal in "\ ". Prefer explicit escaping: "\\ ".
  ^-- SC1117: Backslash is literal in "\ ". Prefer explicit escaping: "\\ ".


In ./helpers/setup-sa.sh line 172:
\ \ role: roles/billing.user" policy-tmp-$$.yml && rm policy-tmp-$$.yml.bak
^-- SC1117: Backslash is literal in "\ ". Prefer explicit escaping: "\\ ".
  ^-- SC1117: Backslash is literal in "\ ". Prefer explicit escaping: "\\ ".
```